src/nvim/event/socket.c

```diff
@@ -90,14 +71,38 @@ int socket_watcher_start(SocketWatcher *watcher, int backlog, socket_cb cb)
   int result;
 
   if (watcher->stream->type == UV_TCP) {
-    result = uv_tcp_bind(&watcher->uv.tcp.handle,
-                         (const struct sockaddr *)&watcher->uv.tcp.addr, 0);
+    struct addrinfo *ai = watcher->uv.tcp.addrinfo;
```

Comment:
Also on Windows UV_EINVAL and -EINVAL are going to be different AFAIK. And you return UV errors in other places. Also UV_EACCESS and not `-EACCESS` below, UV_ENOENT and not `-ENOENT`.